src/components/TaskBoardAction/index.jsx

```diff
@@ -1,14 +1,22 @@
 import PropTypes from "prop-types";
 import Button from "@components/ui/form/button";
+import { useDispatch } from "react-redux";
+import { filterTask } from "@features/filter/filterSlice";
 
 function TaskBoardAction({ onTaskCreation }) {
+    const dispatch = useDispatch();
+
+    const handleFilterClick = (filterState) => {
+        dispatch(filterTask(filterState));
+    };
+
     return (
         <section className="task-board__action-wrapper">
             <Button onClick={onTaskCreation}>Create</Button>
             <div className="task-board__filter-buttons">
-                <Button>All</Button>
-                <Button>Incomplete</Button>
-                <Button>Complete</Button>
+                <Button onClick={() => handleFilterClick("all")}>All</Button>
+                <Button onClick={() => handleFilterClick("incomplete")}>Incomplete</Button>
+                <Button onClick={() => handleFilterClick("complete")}>Complete</Button>
```

Comment:
can we use constant here?

Reply:
I've used constants [here](https://github.com/sse-wtag/todo-app-react/pull/6/commits/d54a3f8cb401e89ed509dbb54ca0986a396682ff). Please re-review at your convenience.